proxy/reverse.go

```diff
@@ -203,3 +209,33 @@ func maybeSetForwardedFor(req *http.Request) {
 	}
 	req.Header.Set("X-Forwarded-For", clientIP)
 }
+
+// checkProxyLoop detects a proxy loop by passing around proxyID
+// and endpoints, and call refresh methods to update endpoints
+// if there's any proxy loop found. It returns true if it has
+// found a proxy loop from a endpoint string.
+func (p *reverseProxy) checkProxyLoop(req *http.Request, ep string) bool {
```

Comment:
+1 for splitting into multiple functions. I will change that. And also I was just changing the argument type just to `url.URL` to make it consistent with `redirectRequest`. Thanks